tests/attacks/poison/test_clean_label_backdoor_attack.py

```diff
@@ -30,13 +30,27 @@
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.skip_framework("non_dl_frameworks", "pytorch", "mxnet")
-def test_poison(art_warning, get_default_mnist_subset, image_dl_estimator):
+@pytest.mark.skip_framework("non_dl_frameworks", "mxnet")
+def test_poison(art_warning, get_default_mnist_subset, image_dl_estimator, framework):
     try:
         (x_train, y_train), (_, _) = get_default_mnist_subset
         classifier, _ = image_dl_estimator()
         target = to_categorical([9], 10)[0]
-        backdoor = PoisoningAttackBackdoor(add_pattern_bd)
+        if framework in ["pytorch", "huggingface"]:
```

Comment:
- I think preserving dtype should be integrated into `add_pattern_bd`.
- The setting of `channels_first` could be set by `channels_first=classifier.channels_first`?

Reply:
updated